action.yml

```
@@ -0,0 +1,50 @@
+name: 'Current projects'
+description: |
+  Github Action to determine all projects a pull request has cards on.
+
+  Returns JSON like this:
+
+    [
+      {
+        "databaseId": 123,
+        "name": "my-github-project"
+      }
+    ]
+
+outputs:
+  current-projects:
+    description: "Projects a pull request has cards on"
+    value: ${{ steps.get-current-projects.outputs.current-projects }}
+
+runs:
+  using: composite
+
+  steps:
+    - id: get-current-projects
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      shell: bash
+      run: |
+        CURRENT_PROJECTS=$(
+          hub api --paginate graphql -f query='
+            query {
+              repository(owner: "${{ github.event.repository.owner.login }}", name: "${{ github.event.repository.name }}") {
+                pullRequest(number: ${{ github.event.pull_request.number }}) {
+                  projectCards {
+                    nodes {
+                      project {
+                        name
+                      }
+                      databaseId
+                    }
+                  }
+                }
+              }
+            }' | jq -c '
+                .data.repository.pullRequest.projectCards.nodes
+              | map({
+                databaseId,
+                name: .project.name
+              })'
+        )
```

let's unify the pipes (e.g. always prefix, not infix) and the brackets (incl. multi-line quotes, e.g. closing bracket/quote always in the same column where the command with the opening bracket/quote starts):

```suggestion
        CURRENT_PROJECTS=$(
          hub api --paginate graphql -f query='
            query {
              repository(owner: "${{ github.event.repository.owner.login }}", name: "${{ github.event.repository.name }}") {
                pullRequest(number: ${{ github.event.pull_request.number }}) {
                  projectCards {
                    nodes {
                      project {
                        name
                      }
                      databaseId
                    }
                  }
                }
              }
            }
          ' \
          | jq -c '
              .data.repository.pullRequest.projectCards.nodes
              | map({
                  databaseId,
                  name: .project.name
                })
            '
        )
```